### PR TITLE
Fix compiling with use_mingw flag on Windows

### PR DESCRIPTION
**Pull request by marcelofg55**
Fixes this bug:
screenshot

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -87,6 +87,7 @@ env_base.split_drivers = False
 env_base.split_modules = False
 env_base.module_version_string = ""
 env_base.msvc = False
+env_base.use_mingw = ARGUMENTS.get("use_mingw", False)
```
Comment (akien-mga) [inline SConstruct]: Why add it to the base env? It's defined as platform-specific option at https://github.com/godotengine/godot/blob/fb29d54dc7301dae422437ad68b0a53c68cc404f/platform/windows/detect.py#L68, so you can just query `env["use_mingw"]`.

File: platform/windows/detect.py [hunk 2]
```diff
@@ -347,12 +347,12 @@ def configure(env):
         env['ENV']['TMP'] = os.environ['TMP']
 
     # First figure out which compiler, version, and target arch we're using
-    if os.getenv("VCINSTALLDIR"):
+    if os.getenv("VCINSTALLDIR") and env.use_mingw == False:
```
Comment (akien-mga) [inline platform/windows/detect.py]: As per above comment, could simply be `!env["use_mingw"]`.

**Comment by akien-mga** [comment]
Maybe also remove the `use_mingw` definition from `platform/uwp/detect.py` as it's unused currently.

File: platform/windows/detect.py (hunk 2 above)
Comment (marcelofg55) [reply platform/windows/detect.py]: Changed

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**